scripts/scripts_utils.py

```diff
@@ -85,69 +106,93 @@ def _download(url: str, local_path: Path) -> Optional[int]:
-        local_path = cache_dir.joinpath(f"{release.value}{ext}")
+        local_path = cache_dir.joinpath(name)
         if local_path.is_file() and want_hash == _get_hash(local_path):
             return str(local_path)
 
         # Download the file.
-        url = f"{_URL}/{release.value}-{version}{ext}"
         retries = 5
         while True:
             err = _download(url, local_path)
             if err is None:
                 break
             retries -= 1
             if retries == 0:
-                exit(
-                    f"Failed to download {release.value}-{version}: HTTP {err}."
-                )
+                exit(f"Failed to download {url}: HTTP {err}.")
             time.sleep(1)
         local_path.chmod(0o755)
 
         # Verify the downloaded hash.
         found_hash = _get_hash(local_path)
         if want_hash != found_hash:
             exit(
-                f"Downloaded {release.value}-{version} but found sha256 "
+                f"Downloaded {url} but found sha256 "
                 f"{found_hash} ({local_path.stat().st_size} bytes), wanted "
                 f"{want_hash}"
             )
 
     return str(local_path)
 
 
+def _get_machine() -> str:
+    machine = platform.machine()
+    if machine == "x86_64":
+        machine = "amd64"
+    return machine
+
+
+def _get_platform_ext() -> str:
+    return ".exe" if platform.system() == "Windows" else ""
+
+
+def get_target_determinator() -> str:
+    """Install the Bazel target-determinator tool to carbon-lang's cache."""
+    # Translate platform information into this tool's release binary form.
+    version = f"{platform.system().lower()}.{_get_machine()}"
+    ext = _get_platform_ext()
+    url = f"{_TARGET_DETERMINATOR_URL}/target-determinator.{version}{ext}"
+
+    # Ensure the platform is supported, and grab its hash.
+    if version not in _TARGET_DETERMINATOR_SHAS:
+        # If this because a platform support issue, we may need to print errors.
+        exit(
+            f"No target-determinator release available for platform: {version}"
+        )
+    want_hash = _TARGET_DETERMINATOR_SHAS[version]
+
+    return _get_cached_binary(f"target-determinator{ext}", url, want_hash)
+
+
+def get_release(release: Release) -> str:
+    """Install a Bazel-released tool to carbon-lang's cache.
+
+    release: The release to cache.
+    """
+    # Translate platform information into Bazel's release form.
+    version = f"{platform.system().lower()}-{_get_machine()}"
+    ext = _get_platform_ext()
+    url = f"{_BAZEL_TOOLS_URL}/{release.value}-{version}{ext}"
+
+    # Ensure the platform is supported, and grab its hash.
+    if version not in _BAZEL_TOOLS_VERSION_SHAS[release.value]:
+        # If this because a platform support issue, we may need to print errors.
+        exit(f"No {release.value} release available for platform: {version}")
+    want_hash = _BAZEL_TOOLS_VERSION_SHAS[release.value][version]
```

Comment:
To the point of helpers, this seems like it could be a helper that takes in the sha list + version and returns the hash. You could also pass a label. The amount of code seems dense though to not have a helper, particularly in the context of something like _get_platform_ext()

Reply:
Ah, sure. Done.